### PR TITLE
 Fix demo_inference to properly normalize input

### PR DESCRIPTION
**Pull request by alexgorban**
Before the fix the demo_inference.py used batch_norm and it did the normalization of input image implicitly. If at the inference time the batch_norm was disabled the inference produced incorrect results.

This fix:
1. does the proper input image normalization and disables the batch_norm at inference time. 
2. introduces demo_inference_test.py to make sure the demo_inference works correctly
3. numerous auto formatting changes.

### CODE REVIEW

**Comment by alexgorban** [comment]
@sguada Can you please review the change?

File: research/attention_ocr/python/demo_inference.py [hunk 1]
```diff
@@ -53,35 +54,40 @@ def load_images(file_pattern, batch_size, dataset_name):
...
-  images_data = load_images(FLAGS.image_path_pattern, FLAGS.batch_size,
-                            FLAGS.dataset_name)
-  with tf.Session() as sess:
-    tf.tables_initializer().run()  # required by the CharsetMapper
-    init_fn(sess)
-    predictions = sess.run(endpoints.predicted_text,
-                           feed_dict={images_placeholder: images_data})
   print("Predicted strings:")
-  for line in predictions:
+  for line in run(FLAGS.checkpoint, FLAGS.batch_size, FLAGS.dataset_name,
```
Comment (sguada) [inline research/attention_ocr/python/demo_inference.py]: Change to:
predictions = run(FLAGS.checkpoint, FLAGS.batch_size, FLAGS.dataset_name, FLAGS.image_path_pattern)
for line in predictions:
 print(line)

Comment (sguada) [reply research/attention_ocr/python/demo_inference.py]: Needed

Comment (alexgorban) [reply research/attention_ocr/python/demo_inference.py]: Done.

**Comment by alexgorban** [comment]
This PR was messed up with unsuccessful merges with the upstream. I will create a new one with the required changes to the Attention OCR only.

**Pull request closed without merging**